.github/scripts/trymerge.py

```diff
@@ -652,6 +654,25 @@ def merge_into(self, repo: GitRepo, *, force: bool = False, dry_run: bool = Fals
 
         repo.push(self.default_branch(), dry_run)
 
+    def merge_on_green(self, repo: GitRepo, org: str, project: str, pr_num: int, *, dry_run: bool = False) -> None:
+        start_time = time.time()
+
+        while True:
```

Comment:
Wait does this mean that the merge job will just continually run until the conditions are met?

Shouldn't we just base the merge on green logic on webhook events to make it simpler?

Reply:
Also for some of the attempted merges couldn't we potentially get into land races with other PRs?

I'm also skeptical if Github would accept a push that's out of date with the current master branch as well

Reply:
> Wait does this mean that the merge job will just continually run until the conditions are met?
Yes that's the plan is to just continually call try-merge until we get it working

> Shouldn't we just base the merge on green logic on webhook events to make it simpler?
that was  the original plan and I implemented that in here but got some feedback to try to just put it in here so that it's all in one place to keep it more simple. https://github.com/pytorch/test-infra/pull/302